clojure-ts-mode.el

```diff
@@ -977,18 +983,54 @@ forms like deftype, defrecord, reify, proxy, etc."
     (and prev-sibling
          (clojure-ts--metadata-node-p prev-sibling))))
 
+(defun clojure-ts--anchor-parent-skip-metadata (_node parent _bol)
+  "Anchor function that returns position of PARENT start for NODE.
+
+If PARENT has optional metadata we skip it and return starting position
+of the first child's opening paren.
+
+NOTE: This anchor is used to fix indentation issue for forms with type
+hints."
+  (let ((first-child (treesit-node-child parent 0 t)))
+    (if (clojure-ts--metadata-node-p first-child)
+        ;; We don't need named node here
+        (treesit-node-start (treesit-node-child parent 1))
+      (treesit-node-start parent))))
+
+(defun clojure-ts--match-collection-item-with-metadata (node-type)
+  "Returns a matcher for a collection item with metadata by NODE-TYPE.
+
+The returned matcher accepts NODE, PARENT and BOL and returns true only
+if NODE has metadata and its parent has type NODE-TYPE."
+  (lambda (node _parent _bol)
+    (string-equal node-type
+                  (treesit-node-type
+                   (clojure-ts--node-with-metadata-parent node)))))
+
 (defun clojure-ts--semantic-indent-rules ()
   "Return a list of indentation rules for `treesit-simple-indent-rules'."
   `((clojure
      ((parent-is "source") parent-bol 0)
      (clojure-ts--match-docstring parent 0)
      ;; https://guide.clojure.style/#body-indentation
      (clojure-ts--match-method-body parent 2)
-     (clojure-ts--match-form-body parent 2)
+     (clojure-ts--match-form-body clojure-ts--anchor-parent-skip-metadata 2)
      ;; https://guide.clojure.style/#threading-macros-alignment
      (clojure-ts--match-threading-macro-arg prev-sibling 0)
      ;; https://guide.clojure.style/#vertically-align-fn-args
      (clojure-ts--match-function-call-arg (nth-sibling 2 nil) 0)
+     ;; Collections items with metadata.
+     ;;
+     ;; This should be before `clojure-ts--match-with-metadata', otherwise they
+     ;; will never be matched.
+     (,(clojure-ts--match-collection-item-with-metadata "vec_lit") grand-parent 1)
+     (,(clojure-ts--match-collection-item-with-metadata "map_lit") grand-parent 1)
+     (,(clojure-ts--match-collection-item-with-metadata "set_lit") grand-parent 2)
+     ;;
+     ;; If we enable this rule for lists, it will break many things.
```

Comment:
You might want to mention this in the caveats section of the README, so users are aware we treat lists differently intentionally.

Reply:
Good idea. Done!